lib/iex/lib/iex/server.ex

```diff
@@ -82,7 +82,8 @@ defmodule IEx.Server do
               {^ref, nil} ->
                 {:error, :refused}
               {^ref, leader} ->
-                IEx.Evaluator.start(server, leader)
+                {_, evaluator_state} = run_state(opts)
```

Comment:
This is changing the semantics of the code because `run_state` will now run twice, once here and another one inside IEx.Server run. My recommendation would be to move the whole run_state function to the Evaluator because building the state of the server is now very straight-forward. Make the evaluator receive the `opts`, instead of the state, and build everything in there.